src/test/java/org/folio/controller/KafkaEventListenerTest.java

```diff
@@ -1,108 +1,121 @@
 package org.folio.controller;
 
-import java.util.Map;
-import java.util.Optional;
-import java.util.concurrent.TimeUnit;
-
-import org.apache.kafka.clients.admin.AdminClient;
-import org.apache.kafka.clients.admin.KafkaAdminClient;
-import org.apache.kafka.clients.consumer.ConsumerConfig;
-import org.apache.kafka.clients.consumer.OffsetAndMetadata;
-import org.apache.kafka.clients.producer.KafkaProducer;
-import org.apache.kafka.clients.producer.ProducerConfig;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
 import org.apache.kafka.clients.producer.ProducerRecord;
-import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.common.serialization.StringSerializer;
-import org.awaitility.Awaitility;
-import org.junit.jupiter.api.AfterAll;
+import org.folio.api.BaseIT;
+import org.folio.listener.kafka.KafkaEventListener;
+import org.folio.repository.EcsTlrRepository;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.DynamicPropertyRegistry;
-import org.springframework.test.context.DynamicPropertySource;
-import org.testcontainers.containers.KafkaContainer;
-import org.testcontainers.junit.jupiter.Container;
-import org.testcontainers.junit.jupiter.Testcontainers;
-import org.testcontainers.utility.DockerImageName;
-
-import lombok.SneakyThrows;
-
-@SpringBootTest
-@Testcontainers
-class KafkaEventListenerTest {
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import static org.awaitility.Awaitility.await;
+import static org.folio.support.MockDataUtils.ITEM_ID;
+import static org.folio.support.MockDataUtils.SECONDARY_REQUEST_ID;
+import static org.folio.support.MockDataUtils.getEcsTlrEntity;
+import static org.folio.support.MockDataUtils.getMockDataAsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+
+@EmbeddedKafka(
```

Comment:
No that long ago, we intentionally replaced embedded solutions with test-containers in all of our modules